chart/openfaas/templates/prometheus-rbac.yaml

```diff
@@ -50,6 +50,7 @@ subjects:
 - kind: ServiceAccount
   name: {{ .Release.Name }}-prometheus
   namespace: {{ .Release.Namespace | quote }}
+{{- if ne $functionNs (.Release.Namespace | toString) }}
```

Comment:
What does this line do?

Reply:
This line will check if the functionNs variable is not equal to release namespace. It will execute the role binding only   when they are not same.

The value to variable functionNs is assigned in first line in prometheus-rbac.yaml, it uses Default to assign either of Release NameSpace or function Namespace. 

{{- $functionNs := default .Release.Namespace .Values.functionNamespace }}

Since, in our case if we don't define function Namesapce, it will default to Release Namesapce.

Reply:
I am having trouble understanding how this works because the way it reads in the diff is that you will always print the line above:

```
  namespace: {{ .Release.Namespace | quote }}
```

Then sometimes you will quote the namespace into the YAML just floating below it like:

```
  namespace: {{ .Release.Namespace | quote }}
openfaas-fn
```

That seems like invalid YAML?

Reply:
OK. I think I get it now.

You are skipping this role completely, but do you have the equivalent change for the operator/controller that was using this role?

Reply:
@alexellis  We won't require any change in operator as this role will exist in operator namespace(openfaas).

In YAML, two role bindings with the same name(openfaas-prometheus) are created in Operator namespace(openfaas) and functional namespace(openfaas-fn). The YAML will fail to apply only when Operator Namespace = Functional namespace, during which we will skip duplicate role execution.